src/gui/mainwindow.cpp

```diff
@@ -695,6 +697,25 @@ void MainWindow::displayRSSTab(bool enable)
     }
 }
 
+void MainWindow::showFilterContextMenu(const QPoint &)
+{
+    Preferences *const pref = Preferences::instance();
+
+    QMenu *menu = m_searchFilter->createStandardContextMenu();
+    menu->addSeparator();
+    QAction *useRegexAct = new QAction(tr("Use regular expressions"), menu);
+    useRegexAct->setCheckable(true);
+    bool useRegexAsFilteringPattern = pref->getRegexAsFilteringPattern();
+    useRegexAct->setChecked(useRegexAsFilteringPattern);
+    menu->addAction(useRegexAct);
+
+    connect(useRegexAct, &QAction::toggled, pref, &Preferences::setRegexAsFilteringPattern);
+
+    menu->exec(QCursor::pos());
+    if (useRegexAsFilteringPattern != useRegexAct->isChecked())
+        m_transferListWidget->applyNameFilter(m_searchFilter->text());
```

Comment:
I would replace the if block with this: `connect(useRegexAct, &QAction::toggled, this, [this]() { m_transferListWidget->applyNameFilter(m_searchFilter->text()); });`

Reply:
Nope :) it works now.